src/main.rs

```diff
@@ -123,17 +123,22 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     // update balances
     let print_balances = opt.print_balances;
     loop {
-        futures::executor::block_on(monitor.do_with_balances(|address, token, balance| {
+        futures::executor::block_on(monitor.do_with_balances(|params| {
             if print_balances {
-                print_balance(address, token, &balance);
+                print_balance(params.address_name, params.token_name, &params.balance);
             }
-            match balance {
+            match params.balance {
                 Ok(balance) => balance_metric
-                    .with_label_values(&[address, token])
+                    .with_label_values(&[
+                        params.address_name,
+                        params.token_name,
+                        // this seems to be the only way to print the full address
+                        &format!("{:?}", params.address),
```

Comment:
It implements Lower/UpperHex so you should be able to print it like here: https://doc.rust-lang.org/nightly/core/fmt/trait.LowerHex.html